src/components/admin/SearchableSelector.jsx

```diff
@@ -0,0 +1,125 @@
+/* eslint react/jsx-no-bind: 0 */
+
+import React from 'react';
+import update from 'react-addons-update';
+import _ from 'lodash';
+
+import BaseComponent from 'lib/BaseComponent';
+import SearchBar from 'components/admin/SearchBar';
+
+
+// material-ui
+import Chip from 'material-ui/Chip';
+
+class ObjectChip extends BaseComponent {
+  constructor(props) {
+    super(props);
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick() {
+    this.props.onDelete(this.props.id);
+  }
+
+  render() {
+    return (
+      <Chip
+        onRequestDelete={this.onClick}
+        style={this.props.style}
+      >
+        {this.props.children}
+      </Chip>
+    );
+  }
+}
+
+export default class EditObjectsForm extends BaseComponent {
+  constructor(props) {
+    super(props);
+    this.safeSetState({
+      addObjectValue: '',
+      objectAutocomplete: [],
+    });
+    this.handleDelete = this.handleDelete.bind(this);
+    this.handleClickAdd = this.handleClickAdd.bind(this);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!this.props.disabled &&
+      this.props.objects !== prevProps.objects) {
+      this.props.onChange();
+    }
+  }
+
+  handleClickAdd(object) {
+    // disable this if saving
+    if (this.props.disabled) return;
+
+    const id = object.id;
+    const name = object.name;
+    const alreadyAdded = this.props.objects.find(baseObject => baseObject.id === id) !== undefined;
+
+    if (alreadyAdded) {
+      window.alert('Already added');
```

Comment:
I actually want to maybe make a HOC that enables us to just call a function similarly to this (or somehow modularizing it otherwise), but it instead renders a Material UI Dialogue (that looks much nicer). I'm okay with you merging this in now without it, but we can keep it in mind that we could maybe do this at some point during the process

Reply:
If we want to move to that, it might be worth looking at [SweetAlert](https://sweetalert.js.org/`)

Reply:
I'm going to leave this for later as I'm currently blocking Fred.

Reply:
Yeah no I agree with that.

But hey, SweetAlert looks pretty awesome